src/client.lisp

```diff
@@ -65,23 +65,33 @@
 
 
 (defun prepare-rpc-call-args (args)
-  (cond
-    ((null args)
-     (alexandria:plist-hash-table (list "*args" '()
-                                        "**kwargs" (make-hash-table :test #'equal))))
-    (t
-     (let ((**kwargs (make-hash-table :test #'equal)))
-       (loop :for arg :in args
-             :for rest := args :then (rest rest)
-             :if (keywordp arg)
-               :return (progn
-                         (loop :for (key val) :on rest :by #'cddr :while val
-                               :do (setf (gethash (sanitize-name key) **kwargs) val))
-                         (alexandria:plist-hash-table (list "*args" *args
-                                                            "**kwargs" **kwargs)
-                                                      :test #'equal))
-             :else
-               :collect arg :into *args)))))
+  (let ((*args (list))
+        (**kwargs (make-hash-table :test #'equal)))
+    (labels
+        ((process-args (args)
+           (cond
+             ((null args)
+              t)
+             ((keywordp (first args))
+              (process-**kwargs args))
+             (t
+              (push (first args) *args)
+              (process-args (rest args)))))
+         
+         (process-**kwargs (args)
+           (cond
+             ((null args)
+              t)
+             (t
+              (let ((key (sanitize-name (first args)))
+                    (val (second args)))
+                (setf (gethash key **kwargs) val)
+                (process-**kwargs args))))))
+      
+      (process-args args)
+      (alexandria:plist-hash-table
```

Comment:
you need to make sure this is an `'equal` hash table

Reply:
Good eye; done.